src/vs/base/browser/ui/menu/menu.ts

```diff
@@ -70,4 +106,139 @@ export class Menu {
 			this.listener = null;
 		}
 	}
+}
+
+class MenuActionItem extends ActionItem {
+	static MNEMONIC_REGEX: RegExp = /&&(.)/g;
+
+	constructor(ctx: any, action: IAction, options: IActionItemOptions = {}) {
+		options.isMenu = true;
+		super(action, action, options);
+	}
+
+	private _addMnemonic(action: IAction, actionItemElement: HTMLElement): void {
+		let matches = MenuActionItem.MNEMONIC_REGEX.exec(action.label);
+		if (matches && matches.length === 2) {
+			let mnemonic = matches[1];
+
+			let ariaLabel = action.label.replace(MenuActionItem.MNEMONIC_REGEX, mnemonic);
+
+			actionItemElement.accessKey = mnemonic.toLocaleLowerCase();
+			this.$e.attr('aria-label', ariaLabel);
+		} else {
+			this.$e.attr('aria-label', action.label);
+		}
+	}
+
+	public render(container: HTMLElement): void {
+		super.render(container);
+
+		this._addMnemonic(this.getAction(), container);
+		this.$e.attr('role', 'menuitem');
+	}
+
+	public _updateLabel(): void {
+		if (this.options.label) {
+			let label = this.getAction().label;
+			if (label && this.options.isMenu) {
+				label = label.replace(MenuActionItem.MNEMONIC_REGEX, '$1\u0332');
+			}
+			this.$e.text(label);
+		}
+	}
+}
+
+class SubmenuActionItem extends MenuActionItem {
+	private mysubmenu: Menu;
+
+	constructor(
+		action: IAction,
+		private submenuActions: IAction[],
+		private parentData: ISubMenuData,
+		private submenuOptions?: IMenuOptions
+	) {
+		super(action, action, { label: true, isMenu: true });
+	}
+
+	public render(container: HTMLElement): void {
+		super.render(container);
+
+		this.builder = $(container);
+		$(this.builder).addClass('monaco-submenu-item');
+		$('span.submenu-indicator').text('\u25B6').appendTo(this.builder);
+		this.$e.attr('role', 'menu');
+
+		$(this.builder).on(EventType.KEY_UP, (e) => {
+			let event = new StandardKeyboardEvent(e as KeyboardEvent);
+			if (event.equals(KeyCode.RightArrow)) {
+				EventHelper.stop(e, true);
+
+				this.createSubmenu();
+			}
+		});
+
+		$(this.builder).on(EventType.KEY_DOWN, (e) => {
+			let event = new StandardKeyboardEvent(e as KeyboardEvent);
+			if (event.equals(KeyCode.RightArrow)) {
+				EventHelper.stop(e, true);
+			}
+		});
+
+		$(this.builder).on(EventType.MOUSE_OVER, (e) => {
+			this.cleanupExistingSubmenu(false);
+			this.createSubmenu();
+		});
+
+
+		$(this.builder).getHTMLElement().onmouseleave = (evt: MouseEvent) => {
```

Comment:
@sbatten this should be exposed if missing so that it reads nicer (avoid the `getHTMLElement()`)

Reply:
@bpasero clarifying... modify the builder api to support onmouseleave?

Reply:
@sbatten so that you can write:

`$(this.builder).on(EventType.MOUSE_LEAVE,...`